File: 167 Two Sum ii.py
#Leetcode link: https://leetcode.com/problems/two-sum-ii-input-array-is-sorted/
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        for i in range(len(numbers)):
            l, r = i + 1, len(numbers) - 1
            tmp = target - numbers[i]
            while l <= r:
                mid = l + (r - l)//2
                if numbers[mid] == tmp:
                    return [i + 1, mid + 1]
                elif numbers[mid] < tmp:
                    l = mid + 1
                else:
                    r = mid - 1
        return []

File: 167 Two Sum ii.swift
//Leetcode link: https://leetcode.com/problems/two-sum-ii-input-array-is-sorted/

class Solution {
    func twoSum(_ numbers: [Int], _ target: Int) -> [Int] {
        for i in 0..<numbers.count {
            var l = i + 1, r = numbers.count - 1
            let tmp = target - numbers[i]

            while l <= r {
                let mid = l + (r - l) / 2
                if numbers[mid] == tmp {
                    return [i + 1, mid + 1]
                } else if numbers[mid] < tmp {
                    l = mid + 1
                } else {
                    r = mid - 1
                }
            }
        }
        return []
    }
}
